frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
+
+            // Fetch the position related to other instructor
+            resp = await apiGET(`/admin/instructors`);
+            const otherInstructor = resp.payload.find(
+                (instructor) =>
+                    instructor.utorid ===
+                    databaseSeeder.seededData.instructors[4].utorid
+            );
+            expect(otherInstructor).toBeDefined();
+            resp = await apiGET(`/admin/sessions/${session.id}/positions`);
+            const otherPosition = resp.payload.find((position) =>
+                position.instructor_ids.find((id) => id === otherInstructor.id)
+            );
+            expect(otherPosition).toBeDefined();
+
+            // Create another assignment (with no DDAH)
+            // for position associated with other instructor
+            const otherAssignment = {
+                note: "",
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: otherPosition.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", otherAssignment);
+            expect(resp).toHaveStatus("success");
+            foreignAssignment = resp.payload;
+        });
 
-        // Get newly created instructor and create a DDAH for them
-        resp = await apiGET(`/admin/instructors`);
-        expect(resp).toHaveStatus("success");
-        const instructorId = resp.payload.find(
-            (instructor) => instructor.utorid === instructorUser.utorid
-        )?.id;
-        expect(instructorId).toBeDefined();
-        const newDDAH = await createDdahWithFixedDuties(instructorId);
+        beforeEach(async () => {
+            await switchToInstructorOnlyUser();
+        });
 
-        // Test the DDAH is fetched properly
-        await switchToInstructorOnlyUser();
-        resp = await apiGET(`/instructor/sessions/${session.id}/ddahs`);
-        expect(resp).toHaveStatus("success");
-        expect(resp.payload).toHaveLength(1);
-        expect(resp.payload[0]).toStrictEqual(newDDAH);
-    });
+        it("fetch Ddahs associated with self only", async () => {
+            // Our instructor should only be able to fetch ddahs associated with them
+            const resp = await apiGET(
+                `/instructor/sessions/${session.id}/ddahs`
+            );
+            expect(resp).toHaveStatus("success");
+
+            // Only one DDAH is intially seeded for this instructor
+            expect(resp.payload).toHaveLength(1);
+
+            // We get duties returned sorted in ascending order, so
+            // we need to sort the seeded ones before comparing
+            const sortedSeededDuties = databaseSeeder.seededData.ddahs[0].duties.sort(
+                (first, second) => first.order - second.order
+            );
+            expect(resp.payload[0].duties).toEqual(sortedSeededDuties);
```

Comment:
Better to save the DDAH as a new variable rather than access the payload directly.

Reply:
That's similar to [this approach here](https://github.com/uoft-tapp/tapp/blob/fd163fd2b23401d63931250d1da9f6212634c416/frontend/src/tests/instructor-permission-test.js#L116), so I assumed that would be ok for one-time use, will change.